### PR TITLE
feature: add create roles API

### PR DESCRIPTION
**Pull request by shreyanshJain852**
## Description

Add create roles API
- Add create roles API route
- Handle transaction for the API for updating the database
- Handle the method calls asynchronously
- Update database context namespace

## Screenshots

API route and payload
create-role-payload

Success API response
success-response

Bad request API response
bad-request

## Summary by CodeRabbit

- **New Features**
	- Introduced a new controller for managing access control roles, including creation and module validation.

### CODE REVIEW

File: src/OsmoGuard.Core/Entities/DTOs/RolesDTO/CreateAcmRoleResponseDTO.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+namespace OsmoGuard.Core.Entities.DTOs.RolesDTO;
+
+public class CreateAcmRoleResponseDTO
+{
+    public int Id { get; set; }
+    public string Name { get; set; } = null!;
+    public string? Description { get; set; }
+    public Status Status { get; set; }
+    public CreateAcmModulePrivilegesResponseDTO ModulePrivilege { get; set; }
```
[the remark this answers is not in the document]
Comment (shreyanshJain852) [reply src/OsmoGuard.Core/Entities/DTOs/RolesDTO/CreateAcmRoleResponseDTO.cs:12]: This is a response DTO class and the ModulePrivilege will always be bound in the response

File: src/OsmoGuard.Core/Entities/DTOs/RolesDTO/CreateAcmRoleRequestDTO.cs [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+public class CreateAcmRoleRequestDTO
+{
+
+    [Required(ErrorMessage = "Name is required")]
+    public string Name { get; set; } = null!;
+    public string? Description { get; set; }
+
+    [Required(ErrorMessage = "Module Privileges are required")]
+    public CreateAcmModulePrivilegesRequestDTO ModulePrivilege { get; set; }
```
[the remark this answers is not in the document]
Comment (shreyanshJain852) [reply src/OsmoGuard.Core/Entities/DTOs/RolesDTO/CreateAcmRoleRequestDTO.cs:14]: We have marked ModulePrivilege as a required object for this request, so it cannot be null

**Comment by sameer-s-b** [comment]
While creating a role, the user could be setting the module privileges for multiple modules at once. The screenshot of the request body appears like it is being done for one single module onlyl

**Comment by shreyanshJain852** [comment]
> While creating a role, the user could be setting the module privileges for multiple modules at once. The screenshot of the request body appears like it is being done for one single module onlyl

Updated the controller to accept list of module privileges

**Comment by sameer-s-b** [comment]
Instead of passing "none", "own" and "all" can you return numbers?
Instead of status "Active", please return "true" or "false"

**Comment by shreyanshJain852** [comment]
> Instead of passing "none", "own" and "all" can you return numbers? Instead of status "Active", please return "true" or "false"

Updated module privileges to be handled as int.

The status in the record represents an active or inactive (deleted) state, which is usually not required by the client for any processing.

**Pull request merged**